src/eth_provider/constant.rs

```diff
@@ -20,7 +20,10 @@ pub const ADDRESS_HEX_STRING_LEN: usize = 40;
 /// Starknet Modulus: 0x800000000000011000000000000000000000000000000000000000000000001
 pub const STARKNET_MODULUS: U256 = U256::from_limbs([0x1, 0, 0, 0x0800_0000_0000_0011]);
 /// Maximum number of times a transaction can be retried
-pub const TRANSACTION_MAX_RETRIES: u8 = 10;
+pub const TRANSACTION_MAX_RETRIES: u8 = u8::from_str(
+    &std::env::var("TRANSACTION_MAX_RETRIES")
+        .unwrap_or_else(|_| panic!("Missing environment variable TRANSACTION_MAX_RETRIES"))
+).expect("failing to parse TRANSACTION_MAX_RETRIES");
```

Comment:
You need to import 

```rust
use std::str::FromStr;
```

See compilation error in the CI

Reply:
oh!
didn't noticed, will fix it!
btw, could you address the other comment. I am afraid I discovered another error saying: "cannot call non-const fun <from_str> in constants"
so I will have to move TRANSACTION_MAX_ENTRIES to another file

Reply:
Yes you can use lazy_static for that, see the example of `RETRY_TX_INTERVAL` because here at the moment you are declaring `TRANSACTION_MAX_RETRIES` as a pub const but you are using `from_str` which is not a const function and this is forbidden by Rust to do this.